app/models/comment.rb

```diff
@@ -0,0 +1,13 @@
+class Comment < ApplicationRecord
+  # Associations
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  # Attributes
+  attribute :text, :text
+
+  # Methods
+  def update_post_comments_counter
+    post.update(comments_counter: post.comments.count + 1)
+  end
+end
```

Comment:
- [OPTIONAL] You're on the right track with the `update_post_comments_counter` method in your `Comment` model. To improve your code, I would suggest a couple of changes:

1. **Use `after_save` Callback**: Consider using the `after_save` callback to automatically update the `comments_counter` of the associated post whenever a new comment is created. This ensures that the counter stays in sync without needing to call the method manually.

2. **Make `update_post_comments_counter` Private**: Since `update_post_comments_counter` is an internal method used for the callback, you can make it private to signify that it's not meant to be called externally. This helps keep your code organized and encourages proper usage.

Here's how you can make these changes:

```ruby
class Comment < ApplicationRecord
  # Associations
  belongs_to :author, class_name: 'User'
  belongs_to :post, class_name: 'Post'

  # Attributes
  attribute :text, :text

  # Callbacks
  after_save :update_post_comments_counter

  private

  def update_post_comments_counter
    post.update(comments_counter: post.comments.count)
  end
end
```

By implementing these suggestions, your code becomes more self-contained and follows best practices for managing associations and callbacks in ActiveRecord models. Great job! 🚀📝